app/views/contexts/list_context.py

```diff
@@ -0,0 +1,90 @@
+from functools import partial
+
+from flask import url_for
+from flask_babel import lazy_gettext
+
+from . import Context
+
+
+class ListContext(Context):
+    def __init__(
+        self,
+        language,
+        schema,
+        answer_store,
+        list_store,
+        progress_store,
+        metadata,
+        summary_definition,
+        for_list,
+        return_to=None,
+        edit_block_id=None,
+        remove_block_id=None,
+    ):
+        super().__init__(
+            language, schema, answer_store, list_store, progress_store, metadata
+        )
+
+        self._summary_definition = summary_definition
+        self._for_list = for_list
+        self._return_to = return_to
+        self._edit_block_id = edit_block_id
+        self._remove_block_id = remove_block_id
+        self._editable = self._edit_block_id or self._remove_block_id
+
+        list_items = (
+            list(self._build_list_items_context()) if summary_definition else []
+        )
+
+        self._context = {
+            "list": {
+                "list_items": list_items,
+                "editable": any([edit_block_id, remove_block_id]),
+            }
+        }
+
+    def get_context(self):
+        return self._context
```

Comment:
This is inconsistent with `QuestionnaireSummaryContext` and `SectionSummaryContext` that both use the `__call__` approach. Shouldn't we decide a common approach and use it consistently across all contexts?

Reply:
What's the preference? I prefer call, but wanted to see what other people preferred?

Reply:
Making non-ABC properties part of the call args would mean that we'd need to chain them down through any of the classes methods. I think I commented somewhere to this effect. Not sure of the best solution, maybe the the init initialises the class properties and the call then uses them to build/return the context?

Reply:
I think moving the non-ABC properties into the init has muddied the water. The other contexts (`QuestionnaireSummaryContext` and `SectionSummaryContext`) don't override the init and pass the arguments that could vary between calls via the `__call__` method. I think we should go for consistency, even if it means passing the `__call__` args down through a few methods.

Reply:
Ok - I'll revert the currently overridden init.